tt_metal/impl/kernels/kernel.cpp

```diff
@@ -213,6 +213,8 @@ RuntimeArgsData &Kernel::common_runtime_args_data() { return this->common_runtim
 void Kernel::validate_runtime_args_size(
     size_t num_unique_rt_args, size_t num_common_rt_args, const CoreCoord &logical_core) {
     uint32_t total_rt_args = (num_unique_rt_args + num_common_rt_args);
+    auto arch = hal.get_arch();
```

Comment:
so we have a way to pull the arch out of tt::Cluster, don't think we need to duplicate that

need a separate discussion of what lives in Cluster and what gets pulled out of the HAL. This can be finetuned in separate PR

Reply:
Last night it made sense to me that we should be able to ask the Hal what arch it is abstracting.

Reply:
yeah I do agree. I guess its okay there are multiple ways to access arch